### ISSUE TITLE
Pro-guard Rules are too broad

### ISSUE THREAD

**Issue by ScottPierce** [PDF417/pdf417-android#23]
Please provide more specific pro-guard rules. This is what you've provided:
```
-keep class com.microblink.** { *; }
-keepclassmembers class com.microblink.** { *; }
-dontwarn android.hardware.**
-dontwarn android.support.v4.**
```

These rules essentially just keep everything in the `com.microblink` package.

Keeping your entire library is a bad practice, and defeats the point of running pro-guard. Please be specific about the parts of the library that aren't compatible with pro-guard, provide and a more detailed pro-guard config that allows pro-guard to tree shake / shrink as much as possible.

**Comment by DoDoENT** [comment]
Our library has already been shrinked and optimised by ProGuard (you can check this by decompiling the JAR). Only API and native methods have been left. If you clean those in your app, it will probably crash.

**Comment by DoDoENT** [comment]
Hello!

This issue has been addressed in [v6.4.0](https://github.com/PDF417/pdf417-android/releases/tag/6.4.0).  We now use annotations library to annotate classes that need to be kept by ProGuard. That is much easier and less error-prone than supporting a massive ProGuard exclusion rules file.

If the issue is solved for you now, please close it.
